packages/nimble-components/src/drawer/index.ts

```diff
@@ -92,6 +92,29 @@ export class Drawer<CloseReason = void> extends FoundationElement {
         return true;
     }
 
+    /**
+     * @internal
+     */
+    public closeHandler(): void {
+        if (this.resolveShow) {
+            // If
+            // - the browser implements dialogs with the CloseWatcher API, and
+            // - the user presses ESC without first interacting with the drawer (e.g. clicking, scrolling),
+            // the cancel event is not fired and the drawer just closes.
+            this.notifyClosed(UserDismissed);
```

Comment:
Specifically for the drawer the assumption is that if the close handler is called with resolveShow still defined then the animation did not start right? I want to make sure that there are not animation event listeners, etc. that need to be cleaned up.

If that's true can you mention that assumption in the comment?